index.js

```diff
@@ -157,7 +158,7 @@ function compression(options) {
 
       // compression method
       var accept = accepts(req)
-      var method = accept.encoding(['gzip', 'deflate', 'identity'])
+      var method = accept.encoding(available)
 
       // negotiation failed
       if (!method || method === 'identity') {
```

Comment:
@hex7c0 , right, that is another case that needs to be fixed and why this PR hasn't been merged just yet :)